kubernetes/cache/cache.go

```diff
@@ -178,11 +192,27 @@ func (c *kialiCacheImpl) CheckNamespace(namespace string) bool {
 	return c.isKubernetesSynced(namespace) && c.isIstioSynced(namespace)
 }
 
+// RefreshNamespace will delete the specific namespace's cache and create a new one.
+func (c *kialiCacheImpl) RefreshNamespace(namespace string) {
+	if nsChan, exist := c.stopChan[namespace]; exist {
+		if nsChan != nil {
+			close(nsChan)
+			nsChan = nil //nolint
+		}
+	}
+	defer c.cacheLock.Unlock()
+	c.cacheLock.Lock()
+	delete(c.nsCache, namespace)
+	c.createCache(namespace)
+}
+
 func (c *kialiCacheImpl) Stop() {
 	log.Infof("Stopping Kiali Cache")
-	if c.stopChan != nil {
-		close(c.stopChan)
-		c.stopChan = nil
+	for _, nsChan := range c.stopChan {
+		if nsChan != nil {
+			close(nsChan)
+			nsChan = nil //nolint
```

Comment:
Same here, it would need to get the key to delete entry from map:
```
	for namespace, nsChan := range c.stopChan {
		if nsChan != nil {
			close(nsChan)
			delete(c.stopChan, namespace)
		}
	}
```

Reply:
I think there is also no need to add if nsChan here.